addons/functions/functions/fnc_digFX.sqf

```diff
@@ -0,0 +1,49 @@
+#include "script_component.hpp"
+/*
+ * Author: nomisum
+ * Displaying Dig FX on Digging
+ *
+ * Arguments:
+ * 0: Trench <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [TrenchObj] call grad_trenches_functions_fnc_digFX;
+ *
+ * Public: No
+ */
+
+params ["_vehicle"];
+
+private _count = 0;
+
+switch (typeof _vehicle) do { 
+    case "GRAD_envelope_short" : {  _count = 7; }; 
+    case "ACE_envelope_small" : {  _count = 5; }; 
+    case "ACE_envelope_big" : {  _count = 7; }; 
+    case "GRAD_envelope_giant" : {  _count = 11; }; 
+    case "GRAD_envelope_vehicle" : {  _count = 13; }; 
+    case "GRAD_envelope_long" : {  _count = 15; }; 
+    default {  }; 
+};
+
+for "_i" from 1 to _count do {
```

Comment:
Instead of hard coding values why not dynamically finding the number of selections. 

You could just set a value within the vehicles config to return how many fx selections there are or find the selections in the script:
```sqf
private _selections = (_vehicle selectionNames "Memory") select {(_x find "fxSource") isEqualTo 0};
```
And then iterate the `_selections` with a `forEach` loop.

Reply:
i was about to dynamically detect and then i loaded the wrong mod and didnt find the selectionnames and then i did the other solution to workaround my non existing limitation

Reply:
hätte `isNotEqualTo` sein müssen :reeeee:

Reply:
Whut? Hold up!!

The [find Alternative Syntax](https://community.bistudio.com/wiki/find#Alternative_Syntax) docs state:
> Zero based position of the first sequence of characters that matches substring, -1 if not found

You want to find all selection names, which start `"fxSource"`. Correct?

So the expression `_name find "fxSource"` should return `0` for the items you want to keep -> You use `(_x find "fxSource") isEqualTo 0` ( NOT `isNotEqualTo`) as your condition, because you want to keep all selection names, which start with `"fxSource"`.

Am I getting something wrong here?

Reply:
idk i didnt even think about it but it didnt work and reversing did

Reply:
![](https://i.imgur.com/R5fFtr2.png)

Reply:
![image](https://user-images.githubusercontent.com/12705416/123523574-a67bd700-d6c4-11eb-8700-fa91ad218152.png)

Reply:
I'm pretty sure we want the first one 😅

Reply:
test with the real thing:

![20210626212805_1](https://user-images.githubusercontent.com/500164/123523687-77b23080-d6c5-11eb-8139-f3209389b862.jpg)
![20210626212815_1](https://user-images.githubusercontent.com/500164/123523688-784ac700-d6c5-11eb-93a2-2eccfe0af518.jpg)

Reply:
seems like a case sensitive issue